jsonrpc2/http_test.go

```diff
@@ -139,3 +141,35 @@ func TestHTTPClient(t *testing.T) {
 		t.Errorf("Call(%v) = %v, want = %v", in, got, want)
 	}
 }
+
+type ContentTypeHanlder string
+
+func (h ContentTypeHanlder) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", string(h))
+	io.WriteString(w, `{"jsonrpc":"2.0","id":0,"result":8}`)
+}
+
+func TestHTTPClientContentType(t *testing.T) {
+	const contentType = "application/json"
+	cases := []struct {
+		contentType string
+		errorString string
+	}{
+		{contentType, "<nil>"},
+		{contentType + "; charset=utf-8", "<nil>"},
+		{contentType + "fail", `{"code":-32603,"message":"bad HTTP Content-Type: application/jsonfail"}`},
```

Comment:
This error is generated by encoding struct or map to json, so order of fields in string representation isn't guaranteed. So it's better to use reflect.DeepEqual or decoding error with ServerError() to compare in test.